templates/index.js

```diff
@@ -13,82 +13,64 @@ handlebars.registerHelper(
+      readFile(path.join(__dirname, name + '.txt'), { encoding: 'utf8' }),
+      readFile(path.join(__dirname, name + '.html'), { encoding: 'utf8' })
     ]
   )
   .spread(
-    function (verifyHtml, verifyText, resetHtml, resetText, unlockHtml,
-      unlockText, passwordChangedHtml, passwordChangedText, passwordResetHtml,
-      passwordResetText, newSyncDeviceHtml, newSyncDeviceText) {
-      var renderVerifyHtml = handlebars.compile(verifyHtml)
-      var renderVerifyText = handlebars.compile(verifyText)
-      var renderResetHtml = handlebars.compile(resetHtml)
-      var renderResetText = handlebars.compile(resetText)
-      var renderUnlockHtml = handlebars.compile(unlockHtml)
-      var renderUnlockText = handlebars.compile(unlockText)
-      var renderPasswordChangedHtml = handlebars.compile(passwordChangedHtml)
-      var renderPasswordChangedText = handlebars.compile(passwordChangedText)
-      var renderPasswordResetHtml = handlebars.compile(passwordResetHtml)
-      var renderPasswordResetText = handlebars.compile(passwordResetText)
-      var renderNewSyncDeviceHtml = handlebars.compile(newSyncDeviceHtml)
-      var renderNewSyncDeviceText = handlebars.compile(newSyncDeviceText)
-
+    function (text, html) {
+      var renderText = handlebars.compile(text)
+      var renderHtml = handlebars.compile(html)
       return {
-        verifyEmail: function (values) {
+        name: camelize(name) + 'Email',
+        fn: function (values) {
           return {
-            html: renderVerifyHtml(values),
-            text: renderVerifyText(values)
-          }
-        },
-        recoveryEmail: function (values) {
-          return {
-            html: renderResetHtml(values),
-            text: renderResetText(values)
-          }
-        },
-        unlockEmail: function (values) {
-          return {
-            html: renderUnlockHtml(values),
-            text: renderUnlockText(values)
-          }
-        },
-        passwordChangedEmail: function (values) {
-          return {
-            html: renderPasswordChangedHtml(values),
-            text: renderPasswordChangedText(values)
-          }
-        },
-        passwordResetEmail: function (values) {
-          return {
-            html: renderPasswordResetHtml(values),
-            text: renderPasswordResetText(values)
-          }
-        },
-        newSyncDeviceEmail: function (values) {
-          return {
-            html: renderNewSyncDeviceHtml(values),
-            text: renderNewSyncDeviceText(values)
+            text: renderText(values),
+            html: renderHtml(values)
           }
         }
       }
     }
   )
 }
+
+module.exports = function () {
+  return P.all(
+    [
+      loadTemplates('verify'),
+      loadTemplates('reset'),
+      loadTemplates('unlock'),
+      loadTemplates('password_changed'),
+      loadTemplates('password_reset'),
+      loadTemplates('new_sync_device'),
+      loadTemplates('post_verify')
```

Comment:
This is a sign that we should add some tests, to at least assert we can correctly load and render each type of email; not necessarily in this PR though